GoogleUtilities/Environment/GULHeartbeatDateStorage.m

```diff
@@ -74,59 +72,43 @@ - (NSURL *)directoryPathURL {
  * @param directoryPathURL The path to the directory that needs to exist.
  */
 - (void)checkAndCreateDirectory:(NSURL *)directoryPathURL {
-  NSError *fileCoordinatorError = nil;
-  [self.fileCoordinator
-      coordinateWritingItemAtURL:directoryPathURL
-                         options:0
-                           error:&fileCoordinatorError
-                      byAccessor:^(NSURL *writingDirectoryURL) {
-                        NSError *error;
-                        if (![writingDirectoryURL checkResourceIsReachableAndReturnError:&error]) {
-                          NSError *error;
-                          [[NSFileManager defaultManager] createDirectoryAtURL:writingDirectoryURL
-                                                   withIntermediateDirectories:YES
-                                                                    attributes:nil
-                                                                         error:&error];
-                        }
-                      }];
+  NSError *error;
+  if (![directoryPathURL checkResourceIsReachableAndReturnError:&error]) {
```

Comment:
Does there need to be a lock around the check and directory create, so that two threads don't try to create the directory at the same time?  Or is it safe to ignore the error of the directory already created?

Reply:
The method is currently called only from the sections protected by `@synchronized`, so no race condition should occur. Also, it looks safe to ignore the error of the directory already created. So if I'm not missing anything, we are good here.